services/QuillLMS/app/models/concerns/student.rb

```diff
@@ -69,7 +69,7 @@ def move_activity_sessions(old_classroom, new_classroom)
       else
 
         new_unit_name = "#{name}'s Activities from #{old_classroom.name}"
-        unit = Unit.create(user_id: new_classroom.owner.id, name: new_unit_name)
+        unit = Unit.create!(user_id: new_classroom.owner.id, name: new_unit_name)
```

Comment:
@dandrabik The root issue was that Unit creations were failing silently, which meant that `unit.id` was `nil` on the next line.

The only way a Unit can fail validations is through a unique name collision. 

Would you provide some product direction on next steps? we could:
- concatenate a string onto the name to guarantee uniqueness
- allow the `Unit.create!` exception to be thrown, and delegate next steps to the user. 
- merge this PR as-is, and observe the names associated with the failed controller transactions to see if there's an obvious behavior pattern

Reply:
I think that I would be inclined to add a date/time stamp to the unit name.  It might be worth confirming with Dan G, but I suspect that there are valid reasons to want to move `ActivitySessions` from a classroom to somewhere else more than once.

Reply:
Awaiting reply from Dan G

Reply:
@happythenewsad I would favor creating an auto-incremented retry flow for a limited number of attempts that appends a count to the end, e.g. `"Janes's Activities from Classroom"` then `"Janes's Activities from Classroom 2"` . It's opinionated, but a pattern we use in other places.

Reply:
From Dan G:
> Definitely—seems like a great fix, and there's no reason that activity session name change would negatively impact support/partnerships!

Reply:
Let's add some logic to append a number then. If there's a duplicate, add a 2, else increment from there for a limited amount of numbers (let's say 20).